### PR TITLE
Avoid printing unnecessary message

### PR DESCRIPTION
**Pull request by agahkarakuzu**
FitResultsSave nii or mat attempts to create a dir that already exists when save path is passed as variable. 

This PR introduces a small fix for that.

### CODE REVIEW

**Comment by mathieuboudreau** [comment]
@agahkarakuzu how would you recommend I test this?

**Comment by notZaki** [comment]
Is `folder` guaranteed to be a full path (at least starting with `'./'`) or can it be plain folder name(s)?

I ask because there's an edge case where `~exist(folder, 'dir')` will be false if `folder` matches anything that exists in the search path.
For example, the following code does not create a folder on my machine
```
folder = 'examples/images';
if ~exist(folder,'dir')
    mkdir(folder)
end
```
because one of the entries in my `path` is `/usr/local/MATLAB/R2019b/examples/images`.
It does, however, make the new folders if `folder = './examples/images'`.
Annoyingly, it also makes the new folders if `folder = examples/images/something_else`.

It probably won't be a problem in practice unless the user has multiple project folders appended in `path` and uses the same folder names across projects.
A safer alternative is to use `isfolder()` but that's a recent~ish function.

**Comment by agahkarakuzu** [comment]
@notZaki thanks for the heads up. This feature is mainly to be used with absolute paths. Users are commonly exporting `FitResults` folder in the same directory with their data. I'll remember points you made in managing directories.

**Comment by mathieuboudreau** [comment]
@agahkarakuzu just by looking at the line above, it appears that relative paths are part of the use cases. I.e. if no "folder" variable is passed on, then the relative path "FitResults" inside that folder will be checked.

https://github.com/qMRLab/qMRLab/pull/365/files#diff-53b85909ee84f57bc52012bb5cfb0f46R8-R11

**Comment by agahkarakuzu** [comment]
@mathieuboudreau I would expect to see batch tests fail if there was a problem with the folder creation in that use case. If you can show me an example where it may fail with an example, maybe I can better understand what you mean.

**Comment by mathieuboudreau** [comment]
Sure. 
In you branch,

1. Open MATLAB
2. Open the qMRLab directory and run startup.
3. Create a batch example using the `inversion_recovery` model
4. Open batch example MATLAB file, and replace `FitResultsSave_nii(FitResults);` with `FitResultsSave_nii(FitResults,[],'Test');`
5. Run the batch example. 

This will break because the directory "Test" is already in the path, per what @notZaki mentioned. 

Capture d’écran 2019-10-28 à 14 52 52

Because of how we're dealing with the default case, i.e. if no folder name is passed automatically set it to "FitResults" (per my comment above), then if the user already has a folder called "FitResults" in their path, then they will encounter the same error as I forced above because we're only checking for if the relative folder name "FitResults" exists, which if it's somewhere in their path your fix won't create the directory.

**Comment by mathieuboudreau** [comment]
Note that before running the batch script, I opened the inversion recivery directory so that its the current dir (not the qMRLab)

**Comment by agahkarakuzu** [comment]
@mathieuboudreau this handles those use cases, giving us all three: 

`Test`
`./Test`
`/Somewhere/Over/The/Rainbow/Test/`

**Comment by agahkarakuzu** [comment]
@mathieuboudreau I also pushed one more commit to make bids.utils decode/encode json files using `JSONLab` on `octave`, others are just painful there. Tested on a notebook on Binder with this branch, it works.

File: External/imtool3D_td/External/bids-matlab/+bids/+util/jsonencode.m [hunk 1]
```diff
@@ -54,6 +54,11 @@
         fclose(fid);
     end
     varargout = { txt };
+
+elseif exist('savejson','file') == 2   
+    disp('this');                 % JSONlab
```
Comment (mathieuboudreau) [inline External/imtool3D_td/External/bids-matlab/+bids/+util/jsonencode.m]: Is this line a leftover debugging print?

Comment (agahkarakuzu) [reply External/imtool3D_td/External/bids-matlab/+bids/+util/jsonencode.m]: Yeap, will remove it.

**Comment by agahkarakuzu** [comment]
Okay @mathieuboudreau, I'll insert some comments there.

File: src/Common/FitResultsSave_mat.m [hunk 2]
```diff
@@ -8,11 +8,21 @@ function FitResultsSave_mat(FitResults,folder)
 if ~exist('folder','var'), folder = 'FitResults'; end
 
 if length(strfind(folder,filesep)) <= 1 && length(strfind(folder,['.' filesep]))~=1
+% Normally users should pass folder variable in the ./myDir format to create folder
+% myDir at their current directory. This condition guards against users who pass
+% folder without './' expression preceding the folder name (e.g. myDir or /myDir).
+% The conditional statement checks for the existence of this case, if present, 
+% it prepends absolute path of the pwd to the folder name to ensure that 
+% the folder is created without possibly conflicting with another folder that may 
+% be saved in the MATLAB/Octave's search path (e.g. Test).
```
Comment (mathieuboudreau) [inline src/Common/FitResultsSave_mat.m]: Maybe a bit too much, but it'll do =) Thanks for handling this!

**Comment by mathieuboudreau** [comment]
Approved, I'm going to override and force merge since your last commit was only text, and I know you're in a deadline rush.
